lms/djangoapps/grades/models.py

```diff
@@ -363,3 +365,69 @@ def _prepare_params_visible_blocks_id(cls, params):
         """
         params['visible_blocks_id'] = params['visible_blocks'].hash_value
         del params['visible_blocks']
+
+
+class PersistentCourseGrade(TimeStampedModel):
+    """
+    A django model tracking persistent course grades.
+    """
+
+    class Meta(object):
+        # Indices:
+        # (course_id, user_id) for individual grades
+        # (course_id) for instructors to see all course grades
+        # (user_id) for course dashboard
+        unique_together = [
+            ('course_id', 'user_id'),
+        ]
+
+    # primary key will need to be large for this table
+    id = UnsignedBigIntAutoField(primary_key=True)  # pylint: disable=invalid-name
+
+    # uniquely identify this particular grade object
+    user_id = models.IntegerField(blank=False, db_index=True)
+    course_id = CourseKeyField(blank=False, max_length=255)
+
+    # Information relating to the state of content when grade was calculated
+    subtree_edited_timestamp = models.DateTimeField('last content edit timestamp', blank=False)
+    course_version = models.CharField('guid of latest course version', blank=True, max_length=255)
+    grading_policy_hash = models.CharField('hash of current grading policy', blank=False, max_length=255)
+
+    # Information about the course grade itself
+    percent_grade = models.FloatField(blank=False)
+    letter_grade = models.CharField(u'Letter grade for course', blank=False, max_length=1)
```

Comment:
In the interest of future flexibilty, should this have a longer max_length, in case some user somewhere wants to use +/-, or other non (A, B, C, D, ...) grading systems?